conans/test/conftest.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from conans.errors import ConanException
+
+from conans.client.tools import vswhere, which
+
+tools_available = [
+    'cmake',
+    'gcc', 'clang', 'visual_studio',
+    'autotools', 'pkg_config', 'premake',
+    'file',
+    'git', 'svn',
+]
+
+if not which("cmake"):
+    tools_available.remove("cmake")
+
+if not which("gcc"):
+    tools_available.remove("gcc")
+if not which("clang"):
+    tools_available.remove("clang")
+
+try:
+    vswhere()
+except ConanException:
+    tools_available.remove("visual_studio")
+
+if not which("file"):
+    tools_available.remove("file")
+
+if not which("git"):
+    tools_available.remove("git")
+if not which("svn"):
+    tools_available.remove("svn")
+
+if not which("autoconf") or not which("automake"):
+    tools_available.remove("autotools")
+if not which("pkg-config"):
+    tools_available.remove("pkg_config")
+if not which("premake"):
+    tools_available.remove("premake")
```

Comment:
IMO, it is the CI orchestrator the one that should tell the test_suite which tools are available, we shouldn't detect them in runtime; or we can stop running tests without noticing.

Reply:
Yes, agree, but if I understood it correctly, this PR is the work oriented to classify tests based on their tooling, and make it available for devs. So the syntax to have the CI to force the installed tools should be done in a later PR, this is one with 97 files changed is good as-is in this regard, no need to do everything in the same PR.

Reply:
Then the review is totally different having that in mind, I thought this was related to our CI.

Reply:
I'd say this is a necessary step towards running in our CI, but not yet. Better do it step by step, lets instrument first the test-suite without changing the CI, and we can move from there. At least that was my understanding, please @lasote tell.